### PR TITLE
Add ArrayFieldTemplate implementation

### LINKED ISSUE #4248: Support for arrays in JSONForm
## Is your feature request related to a problem? Please describe.

Currently we don't support array types in the JSONForm. Would be good to.

## Describe the solution you'd like

Implement `ArrayFieldTemplate` for `JSONForm`

## Implementation

Should just be able to return the array of components passed to `ArrayFieldTemplate`

## Describe alternatives you've considered

N/A

## Additional context

N/A

### PR DESCRIPTION
**Pull request by joshxg**
Fixes #4248 

## Change summary

- Adds support for arrays. Didn't add reordering.
- Looks terrible. Not sure how to make it look better :(

## Testing

- [ ] Below schema can be used to test arrays

### Related areas to think about

```
const schema = {
  definitions: {
    patient: {
      type: 'object',
      properties: {
        patientType: {
          title: 'Patient type',
          type: 'string',
          enum: ['Passport', 'Drivers License', 'Other'],
          default: 'Passport',
        },
        patientID: {
          title: 'Patient identification',
          type: 'string',
        },
      },

      dependencies: {
        patientType: {
          oneOf: [
            {
              properties: {
                patientType: {
                  type: 'string',
                  enum: ['Other'],
                },
                otherPatientType: {
                  title: 'What is the identification type?',
                  type: 'string',
                },
              },
            },
            {
              properties: {
                patientType: {
                  type: 'string',
                  enum: ['Passport', 'Drivers License'],
                },
              },
            },
          ],
        },
      },
    },
  },
  type: 'object',
  properties: {
    patientIdentification: {
      type: 'array',
      title: 'Patient identification',
      items: [
        {
          $ref: '#/definitions/patient',
        },
      ],
      additionalItems: {
        $ref: '#/definitions/patient',
      },
    },
  },
};
```

### CODE REVIEW

**Review by katherine-sussol — APPROVED** [review]
Ooh, this is coool, works nicely.

Random initial thoughts on the layout (feel free to ignore in this PR though I guess can revisit later 😆 ):
- Some spacing between these?
image
- Show divider line only in between items (i.e. skip the one at the end?) 
- Could just be me (and I'm no UX expert) but think it might be a bit more intuitive if the remove item button was on the same side of the screen as the add item one?

**Comment by joshxg** [comment]
Thanks @katherine-sussol  - if there's any other ideas.. sturggling hard with the ui- will implement your suggestions!

File: src/widgets/JSONForm/templates/ArrayField.js [hunk 1]
```diff
@@ -1,17 +1,64 @@
...
+const styles = StyleSheet.create({
+  template: {
+    borderWidth: 2,
+    padding: 20,
+    borderRadius: 5,
+    margin: 20,
+    borderColor: SHADOW_BORDER,
+  },
+  divider: { width: '100%', height: 1, marginTop: 20, backgroundColor: DARKER_GREY },
+  text: { fontFamily: APP_FONT_FAMILY, fontSize: 12 },
```
Comment (mark-prins) [inline src/widgets/JSONForm/templates/ArrayField.js:63]: just looks a little better with a gap between button and text
```suggestion
  text: { fontFamily: APP_FONT_FAMILY, fontSize: 12, padding: 5 },
```

**Review by mark-prins — APPROVED** [review]
👍  and 🤯  that's so good!

yes, the look of it could be better - but I don't care.. it works nicely and fixes that little problem beautifully

**Comment by joshxg** [comment]
The minor style updates here: be2db3a

**Pull request merged**